Signal/src/contact/OWSContactsManager.h

```diff
@@ -8,39 +8,44 @@
 #import <SignalServiceKit/PhoneNumber.h>
 #import "CollapsingFutures.h"
 #import "Contact.h"
-#import "ObservableValue.h"
 
 NS_ASSUME_NONNULL_BEGIN
 
-extern NSString *const OWSContactsManagerSignalRecipientsDidChangeNotification;
+extern NSString *const OWSContactsManagerSignalAccountsDidChangeNotification;
 
 @class UIFont;
+@class SignalAccount;
 
 /**
  * Get latest Signal contacts, and be notified when they change.
  */
 @interface OWSContactsManager : NSObject <ContactsManagerProtocol>
 
-@property (nullable, strong) CNContactStore *contactStore;
-@property (nonnull, readonly, strong) NSCache<NSString *, UIImage *> *avatarCache;
+@property (nonnull, readonly) NSCache<NSString *, UIImage *> *avatarCache;
 
-- (nonnull ObservableValue *)getObservableContacts;
+// signalAccountMap and signalAccounts hold the same data.
+// signalAccountMap is for lookup. signalAccounts contains the accounts
+// ordered by display order.
+@property (atomic, readonly) NSDictionary<NSString *, SignalAccount *> *signalAccountMap;
```

Comment:
You're right, they're redundant. `signalAccountMap` is for bulk lookup (e.g. by `ContactsViewHelper`).  `signalAccountForRecipientId:` is redundant but I'm going to leave it in since it has an assert and I think it makes for more readable code.